projects/extension/ai/semantic_catalog.py

```diff
@@ -0,0 +1,383 @@
+import json
+from collections.abc import Generator
+from typing import TypedDict
+
+
+class GeneratedDescription(TypedDict):
+    name: str
+    description: str
+
+
+def render_obj_sample(plpy, relation: str) -> str:
+    result = plpy.execute(f"select * from {relation}", 5)
+
+    ret_obj = f"""<data id="{relation}">"""
+    for r in result:
+        ret_obj += f"""\n  insert into {relation} ({', '.join(r.keys())}) values ({', '.join([str(v) for v in r.values()])});"""
+    ret_obj += "\n</data>"
+
+    return ret_obj
+
+
+def get_parsed_config(plpy, catalog_name: str, config: dict | None) -> dict:
+    if config is None:
+        result = plpy.execute(
+            f"select x.text_to_sql from ai.semantic_catalog x where x.catalog_name = '{catalog_name}';"
+        )
```

Comment:
we should use a parameterized query and pass catalog_name as a parameter

```suggestion
        plan = plpy.plan(
            f"select x.text_to_sql from ai.semantic_catalog x where x.catalog_name = $1;", ["text"]
        )
        result = plpy.execute(plan, [catalog_name], 1)
```

Reply:
Done in 4967a3d51918ff09a597e0f28a1e6daa6a6826a6.

I don't think there's a `plpy.plan` function, used `plpy.prepare` instead.